packages/tools/css-serializer/src/parse.ts

```diff
@@ -280,7 +281,14 @@ export function parse(content: string, options: {
           type: 'StyleRule',
           style: transformBlock(node.block, errors),
           selectorText: {
-            value: preludeText,
+            value: options.enableCSSSelector
+              ? preludeText
+              // When enableCSSSelector is disabled, remove backslash escapes to ensure CSS selector compatibility.
+              // For example, TailwindCSS generates selectors like `.h-\\[370px\\]` which require unescaping.
+              : preludeText
+                .replaceAll('\\[', '[')
+                .replaceAll('\\]', ']')
+                .replaceAll('\\#', '#'),
```

Comment:
Should we also escape `\.`?

E.g.:

```html
<div class="w-[0.6]" />
```

https://play.tailwindcss.com/T1v76iJNt8